osutil/kcmdline.go

```diff
@@ -204,6 +204,115 @@ func KernelCommandLineKeyValues(keys ...string) (map[string]string, error) {
 	return m, nil
 }
 
+// KernelValidateParam validates a kernel parameter and splits it in
+// key and value in case there is no error.
+// See lib/cmdline.c:next_arg in kernel sources.
+
+// KernelArgument represents a parsed kernel argument
+type KernelArgument struct {
+	Param string
+	Value string
+}
+
+// KernelParseCommandline parses a kernel command line, returning a
+// slice with the arguments in the same order as in cmdline. Note that
+// kernel arguments can be repeated. We follow the same algorithm as in
+// the kernel as far as possible.
+// TODO Replace KernelCommandLineSplit with this eventually
+func KernelParseCommandline(cmdline string) (args []KernelArgument) {
+	cmdlineBy := []byte(cmdline)
+	args = []KernelArgument{}
+	start := firstNotSpace(cmdlineBy)
+	for start < len(cmdlineBy) {
+		param, val, end := parseArgument(cmdlineBy[start:])
+		// Hyphens (dashes) and underscores are equivalent in parameter names,
+		// so we standardize in "_".
+		param = strings.ReplaceAll(param, "-", "_")
+		args = append(args, KernelArgument{Param: param, Value: val})
+		start += end
+		start += firstNotSpace(cmdlineBy[start:])
+	}
+
+	return args
+}
+
+// Does the same as isspace() in tools/include/nolibc/ctype.h
```

Comment:
Maybe add "linux kernel" here  ` Does the same as isspace() in the linux kernels tools/include/nolibc/ctype.h` ?

Reply:
Added